src/json.hpp

```diff
@@ -144,6 +337,41 @@ struct DecimalSeparator : std::numpunct<char>
 
 }
 
+// taken from ranges-v3
+// TODO add doc
+template <typename T>
+struct _static_const
```

Comment:
I rechecked and IIUC this usage is fine because the underscore:
- is not followed by another underscore,
- is not followed by a capital letter,

and the identifier is not in the global namespace. 

I am still not sure that I understood the rules correctly though... Avoiding identifiers that start with an underscore was always my pragmatic solution.